### PR TITLE
develop

### PR DESCRIPTION
**Pull request by Julwas**

### CODE REVIEW

File: src/test/java/models/Project.java [hunk 1]
```diff
@@ -3,16 +3,19 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.google.gson.annotations.SerializedName;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.*;
 
+import lombok.*;
 
 @Data
+@Getter
```
Comment (wildpioneer) [inline src/test/java/models/Project.java]: Тут надо прям все аннотации Lombok???
И @Data, и @Getter и  @Builder в одном месте???

File: src/test/java/models/User.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    private String lastName;
+    private String email;
+    private String password;
+    private String confirmPassword;
+
+    @SerializedName(value = "admin")
+    @JsonProperty("admin")
+    @EqualsAndHashCode.Exclude
+    private boolean admin;
+    @SerializedName(value = "id")
```
Comment (wildpioneer) [inline src/test/java/models/User.java]: Форматирование - пустые строки между блоками

File: src/test/java/pages/MyProjectsPage.java [hunk 3]
```diff
@@ -40,26 +56,64 @@ public SelenideElement getButtonCreate() {
...
+    {
+        getManageButton().click();
+        getCreateButton().click();
+    }
+    public void loadFile()
+    {
+        getMyWorkButton().click();
+        getButton().click();
+        String pathToFile = MyProjectsPage.class.getClassLoader().getResource("test.jpg").getPath();
+        getDownloadFileButton().sendKeys(pathToFile.substring(1));
```
Comment (wildpioneer) [inline src/test/java/pages/MyProjectsPage.java]: У меня не работает - хотя пример, как должно быть, я приводил уже 3 раза как

File: src/test/java/tests/api/ApiGetAllProjectsTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+        List<Project> actualProjects = gson.fromJson(respAsJsonArray, listType);
+
+        logger.info("List of all projects: ");
+        for (Project p :
+                actualProjects) {
+            logger.info(p);
+        }
+
+       Assert.assertEquals(actualProjects.get(11), expectedProject);
```
Comment (wildpioneer) [inline src/test/java/tests/api/ApiGetAllProjectsTest.java]: А если добавиться еще проектов - тогда не 11 будет???

File: src/test/java/tests/api/ApiGetProjectTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import static io.restassured.RestAssured.given;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ApiGetProjectTest extends BaseApiTest {
+    static Logger logger = LogManager.getLogger(ApiGetProjectTest.class);
+    private Project actualProject;
+    @Test
```
Comment (wildpioneer) [inline src/test/java/tests/api/ApiGetProjectTest.java:22]: Форматирование

File: src/test/java/tests/api/ApiGetProjectTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+        String pathToFile = ApiGetProjectTest.class.getClassLoader().getResource("expectedProject.json").getPath();
+        FileReader reader = new FileReader(pathToFile);
+
+        Project expectedProject = gson.fromJson(reader, Project.class);
+
+        String json = gson.toJson(expectedProject);
+        System.out.println(json);
+
+        Response response = given()
```
Comment (wildpioneer) [inline src/test/java/tests/api/ApiGetProjectTest.java]: Почему все эти вызовы находятся в тесте а не в адаптере?

File: src/test/java/tests/gui/UITests.java [hunk 7]
```diff
@@ -14,22 +15,22 @@
...
-    @Test(description = "тест на проверку всплывающего сообщения.")
+    @Test(description = "тест на проверку всплывающего сообщения.", priority = 1)
     public void PopUpMessageTest() {
         open(ReadProperties.getUrl());
         loginStep.successLogin(ReadProperties.email(), ReadProperties.password());
 
         myProjectsPage.checkPopUpMessage();
-
         String expectedText = "View comments";
         SelenideElement elementText = $
```
Comment (wildpioneer) [inline src/test/java/tests/gui/UITests.java]: НИкаких использований селекторов в самом тесте быть не должно

File: src/test/java/tests/gui/UITests.java [hunk 8]
```diff
@@ -39,9 +40,10 @@ public void PopUpMessageTest() {
 
         logger.info("PopUpMessageTest. The 'View comments' pop-up message is present.");
     }
+
     @Description("Verifies the process of adding project.")
     @Severity(SeverityLevel.CRITICAL)
-    @Test(description = "Тест на создание сущности")
+    @Test(description = "Тест на создание сущности", priority = 2)
     public void CreationEntityTest  () {
         open(ReadProperties.getUrl());
```
Comment (wildpioneer) [inline src/test/java/tests/gui/UITests.java]: В каждом тесте открывается начальная страница????? Это дубликаты кода - исправить

File: src/test/java/tests/gui/UITests.java [hunk 9]
```diff
@@ -53,7 +55,7 @@ public void CreationEntityTest  () {
     }
     @Description("Checking the input field for boundary values.")
```
Comment (wildpioneer) [inline src/test/java/tests/gui/UITests.java:54]: Форматирование

**Review by wildpioneer — CHANGES_REQUESTED** [review]

File: src/test/java/models/Project.java (hunk 1 above)
Comment (Julwas) [reply src/test/java/models/Project.java]: исправлено

**Comment by Julwas** [comment]
можно проверять

File: src/test/java/models/Project.java (hunk 1 above)
Comment (wildpioneer) [reply src/test/java/models/Project.java]: :) Не угадали 
Надо было @Getter @Builder оставить
